client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -946,6 +948,21 @@ public void close() {
     heartBeatExecutorService.shutdownNow();
     coordinatorClients.forEach(CoordinatorClient::close);
     dataTransferPool.shutdownNow();
+    // wait for all services graceful termination
+    awaitExecutorServiceTermination(5000, heartBeatExecutorService, dataTransferPool, unregisterExecutorService);
```

Comment:
Shall we really wait for all executors or only `unregisterExecutorService`?

Reply:
I think we should wait only for unregisterExecutorService. 
This is because if unregistration fails, it will result in a waste of memory resources on the server side for a period of time. This portion of memory can only be released after timeout.

Reply:
We need to make timeoutMs configurable.

Reply:
Ideally, we would use the timeout of the unregister request. Not sure how easy wiring that through will be.

Reply:
`unregisterRequestTimeSec` is already in `ShuffleWriteClientImpl`. So you can just use it.

Reply:
yeah, that's easy, done!